libs/community/langchain_community/chat_message_histories/falkordb.py

```diff
@@ -0,0 +1,228 @@
+        node_label (str): Label for the session node in the graph. Defaults to "Session".
+        window (int): The number of messages to retrieve when querying the history. Defaults to 3.
+        ssl (bool): Whether to use SSL for connecting to the database. Defaults to False.
+        graph (Optional[FalkorDBGraph]): Optionally provide an existing FalkorDBGraph object for connecting.
+
+    Example:
+        .. code-block:: python
+            from langchain_community.chat_message_histories import FalkorDBChatMessageHistory
+
+            history = FalkorDBChatMessageHistory(
+                session_id="1234",
+                host="localhost",
+                port=6379,
+            )
+            history.add_message(HumanMessage(content="Hello!"))
+    """
+
+    def __init__(
+        self,
+        session_id: Union[str, int],
+        username: Optional[str] = None,
+        password: Optional[str] = None,
+        host: str = "localhost",
+        port: int = 6379,
+        database: Optional[str] = generate_random_string(4),
+        node_label: str = "Session",
+        window: int = 3,
+        ssl: bool = False,
+        *,
+        graph: Optional[FalkorDBGraph] = None,
+    ) -> None:
+        """Initialize the FalkorDBChatMessageHistory class with the session and connection details."""
+        try:
+            import falkordb
+        except ImportError:
+            raise ImportError(
+                "Could not import falkordb python package."
+                "Please install it with `pip install falkordb`."
+            )
+
+        if not session_id:
+            raise ValueError("Please ensure that the session_id parameter is provided.")
+
+        if graph:
+            self._database = graph._graph
+            self._driver = graph._driver
+        else:
+            self._host = host
+            self._port = port
+            self._username = username or os.environ.get("FALKORDB_USERNAME")
+            self._password = password or os.environ.get("FALKORDB_PASSWORD")
+            self._ssl = ssl
+
+            try:
+                self._driver = falkordb.FalkorDB(
+                    host=self._host,
+                    port=self._port,
+                    username=self._username,
+                    password=self._password,
+                    ssl=self._ssl,
+                )
+            except redis.exceptions.ConnectionError:
+                raise ValueError(
+                    "Could not connect to FalkorDB database. "
+                    "Please ensure that the host and port are correct."
+                )
+            except redis.exceptions.AuthenticationError:
+                raise ValueError(
+                    "Could not connect to FalkorDB database. "
+                    "Please ensure that the username and password are correct."
+                )
+
+        self.database_name = database
+        self._database = self._driver.select_graph(database)
```

Comment:
see session_id comment above. if you select graph for each session you use the multi graph ability of falkordb.

Reply:
Done